pkgs/development/interpreters/wamr/default.nix

```diff
@@ -0,0 +1,44 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+}:
+
+stdenv.mkDerivation rec {
+  pname = "wamr";
+  version = "1.2.1";
+
+  src = fetchFromGitHub {
+    owner = "bytecodealliance";
+    repo = "wasm-micro-runtime";
+    rev = "WAMR-${version}";
+    hash = "sha256-RO5GBwiozdMaLt2m6rUlw+UohnYlba6llrVudAsIHW8=";
+  };
+  dontInstall = true;
+
+  nativeBuildInputs = [ cmake ];
+
+  configurePhase = ''
+    runHook preConfigure
+    mkdir -p build
+    cd build
+    cmake ../product-mini/platforms/linux
```

Comment:
Can't you use sourceRoot?

Reply:
I think it's fine now.

Reply:
Using `sourceRoot`, you wouldn't need this custom `configurePhase`, which means you wouldn't need to run `cmake` yourself, which means standard Nixpkgs mechanisms like `cmakeFlags` would work.